packages/react-dev-utils/FlowTypecheckPlugin.js

```diff
@@ -0,0 +1,236 @@
+    newMessage = newMessage.slice(0, stillIndex);
+  }
+  var notRespIndex = newMessage.indexOf("The flow server is not responding");
+  if (notRespIndex >= 0) {
+    newMessage = newMessage.slice(0, notRespIndex);
+  }
+  return newMessage;
+}
+
+function execOneTime(command, args, options) {
+  return new Promise((resolve, reject) => {
+    var stdout = new Buffer("");
+    var stderr = new Buffer("");
+    var oneTimeProcess = childProcess.spawn(
+      command,
+      args,
+      options
+    );
+    oneTimeProcess.stdout.on('data', chunk => {
+      stdout = Buffer.concat([stdout, chunk]);
+    });
+    oneTimeProcess.stderr.on('data', chunk => {
+      stderr = Buffer.concat([stderr, chunk]);
+    });
+    oneTimeProcess.on('error', error => reject(error));
+    oneTimeProcess.on('exit', code => {
+      switch (code) {
+        case 0:
+          return resolve(stdout);
+        default:
+          return reject(new Error(
+            Buffer.concat([stdout, stderr]).toString()
+          ));
+      }
+    });
+  });
+}
+
+function writeFileIfDoesNotExist(path, data) {
+  return new Promise((resolve, reject) => {
+    fs.exists(path, exists => {
+      if (!exists) {
+        fs.writeFile(path, data, err => {
+          if (err) {
+            return reject(err);
+          }
+          resolve(true);
+        });
+      } else {
+        resolve(false);
+      }
+    });
+  });
+}
+
+function writeInFileIfNotPresent(path, contentToAssert, contentToAppend) {
+  return new Promise((resolve, reject) => {
+    fs.exists(path, exists => {
+      if (!exists) {
+        fs.writeFile(path, contentToAppend, err => {
+          if (err) {
+            return reject(err);
+          }
+          resolve(true);
+        });
+      } else {
+        fs.readFile(path, (err, existingContent) => {
+          if (err) {
+            return reject(err);
+          }
+          if (existingContent.indexOf(contentToAssert) < 0) {
+            fs.appendFile(path, contentToAppend, err => {
+              if (err) {
+                return reject(err);
+              }
+              resolve(true);
+            });
+          } else {
+            resolve(false);
+          }
+        });
+      }
+    });
+  });
+}
+
+function getFlowVersion(options) {
+  return execOneTime(
+    (options || {}).global ? "flow" : flowBinPath,
+    ['version', '--json']
+  )
+  .then(rawData => JSON.parse(rawData))
+  .then(versionData => versionData.semver);
+}
+
+function initializeFlow(projectPath, flowconfig, otherFlowTypedDefs) {
+  const flowconfigPath = path.join(projectPath, '.flowconfig');
+  const gitignorePath = path.join(projectPath, '.gitignore');
+  return getFlowVersion().then(localVersion => Promise.all([
+    getFlowVersion({global: true}).catch(() => localVersion)
+    .then(globalVersion =>
+      globalVersion !== localVersion ?
+        Promise.reject(new Error(
+          'Your global flow version does not match react-script\'s flow version.\n\n' +
```

Comment:
@gaearon I know, I couldn't do better than printing a warning. I don't know about nuclide but I know that the [official Flow plugin for VSCode](https://github.com/flowtype/flow-for-vscode) uses the globally installed flow-bin. I am already constantly changing my global flow version when switching projects because of that. If an IDE supports the local flow version, all you have to do is to uninstall the global version. But I prefer keeping the warning though, at first I did not understand why things were so slow, and then I looked at how my processes were spawn and understood...I think that without the warning someone would blame CRA for being slow...